src/Workspaces/Core/Portable/Rename/RenameOptions.cs

```diff
@@ -2,44 +2,33 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
+using System;
+using System.Runtime.Serialization;
 using Microsoft.CodeAnalysis.Options;
 
 namespace Microsoft.CodeAnalysis.Rename
 {
+    [Obsolete("Use SymbolRenameOptions or DocumentRenameOptions instead")]
     public static class RenameOptions
     {
         public static Option<bool> RenameOverloads { get; } = new Option<bool>(nameof(RenameOptions), nameof(RenameOverloads), defaultValue: false);
+
         public static Option<bool> RenameInStrings { get; } = new Option<bool>(nameof(RenameOptions), nameof(RenameInStrings), defaultValue: false);
-        public static Option<bool> RenameInComments { get; } = new Option<bool>(nameof(RenameOptions), nameof(RenameInComments), defaultValue: false);
 
-        /// <summary>
-        /// Set to true if the file name should match the type name after a rename operation
-        /// </summary>
-        internal static Option<bool> RenameFile { get; } = new Option<bool>(nameof(RenameOptions), nameof(RenameFile), defaultValue: false);
+        public static Option<bool> RenameInComments { get; } = new Option<bool>(nameof(RenameOptions), nameof(RenameInComments), defaultValue: false);
 
         public static Option<bool> PreviewChanges { get; } = new Option<bool>(nameof(RenameOptions), nameof(PreviewChanges), defaultValue: false);
     }
 
-    internal readonly record struct RenameOptionSet(
-        bool RenameOverloads,
-        bool RenameInStrings,
-        bool RenameInComments,
-        bool RenameFile)
-    {
-        internal static RenameOptionSet From(Solution solution)
-            => From(solution, options: null);
-
-        internal static RenameOptionSet From(Solution solution, OptionSet options)
-        {
-            options ??= solution.Options;
-
-            return new RenameOptionSet(
-                options.GetOption(RenameOptions.RenameOverloads),
-                options.GetOption(RenameOptions.RenameInStrings),
-                options.GetOption(RenameOptions.RenameInComments),
-                options.GetOption(RenameOptions.RenameFile));
-        }
-    }
+    [DataContract]
+    public readonly record struct SymbolRenameOptions(
+        [property: DataMember(Order = 0)] bool RenameOverloads = false,
+        [property: DataMember(Order = 1)] bool RenameInStrings = false,
+        [property: DataMember(Order = 2)] bool RenameInComments = false,
+        [property: DataMember(Order = 3)] bool RenameFile = false);
+
+    [DataContract]
+    public readonly record struct DocumentRenameOptions(
+        [property: DataMember(Order = 0)] bool RenameMatchingTypeInStrings = false,
+        [property: DataMember(Order = 1)] bool RenameMatchingTypeInComments = false);
```

Comment:
I got confused by this until i understood the direction.  e.g. the top one is if you're renaming a symbol, and then if you want to rename the file, for example.  The bottom one is if you're renaming the document, and you want to rename matches in strings/comments (at least that's how i read it).

Can you have small doc comments explaining this?

Reply:
Sure.